pyvista/core/composite.py

```diff
@@ -132,7 +132,7 @@ def wrap_nested(self):
                 self.SetBlock(i, pyvista.wrap(block))
 
     @property
-    def bounds(self) -> List[float]:
+    def bounds(self) -> tuple:
```

Comment:
I was getting mypy errors when doing that... I'm a typing noob and really don't know

Reply:
Just to be sure: make sure it was `Tuple[float, ...]` and not `Tuple[float]`. Latter means a 1-length tuple. We could probably even specify the length 6 if that helps users during unpacking.

Reply:
I know I tried specifying the length to 6 and ran into issues. I'll try this again and track down/post the exact issue

Reply:
There's also the [mypy playground](https://mypy-play.net/?mypy=latest&python=3.11) for sandboxing.

Reply:
I see at least one of the issues.  The MultiBlock bounds returns some form of `Tuple(ndarray)` like object not a `tuple(float, etc...)`.  I assume we want the typing to be consistent with DataSets here.

Reply:
I think it is compounded by errors in typing in other functions.  Note that this function has ignored typing which is enabling these errors to build up

Reply:
The issue is casting a numpy array to a typed tuple.

MyPy wants this:

```py
tuple(the_bounds)
```

to be this:

```py
(float(the_bounds[0]), float(the_bounds[1]), float(the_bounds[2]), float(the_bounds[3]), float(the_bounds[4]), float(the_bounds[5]))
```

where `the_bounds` is a `ndarray` with `float` dtype.

AFAIK, this might be the only way to get the return type to match `Tuple[float, float, float, float, float, float]`

Reply:
It's not elegant, but I think I'm just going to push this so that it matches the return type of other `bounds()` properties

Reply:
Differently-icky alternative:
```py
from typing import cast
cast(Tuple[float, float, float, float, float, float], tuple(the_bounds))
```

Reply:
Hm. Should I do this instead?

Reply:
Well, this is the use case for `cast` I think (i.e. pinky-swear that something is the correct type). And the manual unpacking is super ugly. The only (very mild) benefit is that if we ever get a shorter array, the current implementation will raise. Mostly academic benefit I think.

Reply:
Agreed. I will use `cast`. Though I usually prefer these sorts of "academic"/theoretical benefits

Reply:
I think the typing is wrong since we sometimes (or always?) return something more like `Tuple[np.floating, np.floating, np.floating, np.floating, np.floating, np.floating]`.  So this may be an opportunity for a future PR to clean up since when I tried to clean it up quickly I ran into typing errors in other functions that interact with this one.

Reply:
I haven't looked closely but a type alias that is a 6-tuple of number-likes would help with that too.

Reply:
Yeah, I'm going to save that for a future PR by someone who knows Python type hinting better than I

Reply:
It should be simple unless it's like a knitted sweater with a loose end, but sure thing.

Reply:
Sometimes its `float` and sometimes its `np.floating`? IMO, `float` should be a sufficient type hint. `np.floating` is only returned when we are computing the `bounds` from many sub meshes or actors (the case for multiblock which is the main thing being addressed in this PR).

Reply:
When I started trying to unravel the thread, I started getting errors in these two functions.  I couldn't figure out where these functions interact with bounds, but there are definite mistakes in the typing here, which is evidenced by ignoring the typing check on one of them.

https://github.com/pyvista/pyvista/blob/993849e84b93adf457e5d4d2372c577c19ed168f/pyvista/core/dataset.py#L794-L796

https://github.com/pyvista/pyvista/blob/993849e84b93adf457e5d4d2372c577c19ed168f/pyvista/core/composite.py#L238

Reply:
I meant basically
https://github.com/pyvista/pyvista/blob/993849e84b93adf457e5d4d2372c577c19ed168f/pyvista/_typing.py#L16
If we get `int`s things should probably still work.